library.ml

```diff
@@ -3,64 +3,77 @@ open Printf
 open String
 open List
 
-class enemy=object (self)
-  val mutable name:string=""
-  val mutable health:int=0
-  val mutable dmg:int=0
-  method set_health n=health <- n
-  method set_name n=name <- n
-  method set_dmg n=dmg <- n
+module Enemy = struct
+  type t =
```

Comment:
`t` is not a magical name, but "culture-wise" it is usually the name given to the main type of a module.
Often it is referred to as the "state" variable of a monadic module, which is what we have here.